### PR TITLE
Code changes for matcher chaining

### LINKED ISSUE #761: Matcher Chaining
With matcher chaining, one can use JSONPath to get a JSON node, and use other matcher to compare its value. For example: 
```
{
  body: [
    {
      matcher: "jsonpath"
      value: “$.user.id"
      doMatch: {
        matcher: “exact”,
        value: “1"
    }
  ]
}
```
This removes the stress of composing and testing complex  jsonpath/xpath expressions, which may not be supported by the current jsonpath/xpath library hoverfly is using. The matcher will be more readable too.

### PR DESCRIPTION
**Pull request by kapishmalik**
Resolves #761. Different way of implementation but following same approach as followed in PR #1035 

PR contains changes for implementing matcher chaining. Any matcher can be combined with any of the other matchers. Output/matched value of one matcher is feed into other matcher mentioned in do match. If this PR looks fine then I will add documentation. 

`{
body: [
{
matcher: "jsonpath"
value: “$.user.id"
domatch: {
matcher: “exact”,
value: “1"
}
]
}

"body": [
{
"matcher": "xpath",
"value": "/root/id",
"domatch": {
"matcher": "exact",
"value": "123"
}
}
]
`

### CODE REVIEW

**Comment by kapishmalik** [comment]
@tommysitu I have added documentation and tested once. You can take a look.

File: core/matching/field_matcher_test.go [hunk 1]
```diff
@@ -155,6 +155,44 @@ var fieldMatcherTests = []fieldMatcherTest{
...
+						DoMatch: &models.RequestFieldMatchers{
+							Matcher: "exact",
+							Value:   "Test",
+						},
+					},
+				},
+			},
+		},
+		toMatch: `<document><details>{"name":"Test", "id":"12345"}</details></document>`,
+		equals:  BeTrue(),
```
Comment (tommysitu) [inline core/matching/field_matcher_test.go:194]: 🤩

**Review by tommysitu — APPROVED** [review]
Beautiful. Much better than the original implementation

File: core/matching/field_matcher_test.go (hunk 1 above)
Comment (kapishmalik) [reply core/matching/field_matcher_test.go:194]: ☺️

**Pull request merged**